opentutor_classifier/opentutor_classifier/lr/clustering_features.py

```diff
@@ -59,38 +60,62 @@ def alignment_metric(self, x, y) -> float:
             self.data[int(y[0])], self.data[int(x[0])]
         )
 
-    def fit_predict(self, data: np.ndarray):
+    def fit_predict(self, data: np.ndarray, train_quality):
         self.data = data
         x = np.arange(len(self.data)).reshape(-1, 1)
 
-        # Calculate pairwise distances with the new metric.
+        # Calculate pairwise distances with the new metric.'
         m = pairwise_distances(x, x, metric=self.alignment_metric)
 
         agg = AgglomerativeClustering(
-            n_clusters=2, affinity="precomputed", linkage="average"
+            n_clusters=train_quality, affinity="precomputed", linkage="average"
         )
         return agg.fit_predict(m)
 
-    def get_clusters(self, good_answers: np.array, bad_answers: np.array):
-        good_labels = self.fit_predict(good_answers)
-        bad_labels = self.fit_predict(bad_answers)
-        return good_labels, bad_labels
+    def get_clusters(
+        self, good_answers: np.array, bad_answers: np.array, train_quality: int
+    ):
+        if train_quality == 1:
```

Comment:
should this be `if train_quality <= 1:` ?

Reply:
train_quality = 0, will not generate features at all.